ksql-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -204,23 +210,33 @@ private ProjectNode buildProjectNode(final PlanNode sourcePlanNode) {
 
     final Optional<ColumnName> keyFieldName = getSelectAliasMatching((expression, alias) ->
         expression instanceof ColumnReferenceExp
-            && ((ColumnReferenceExp) expression).getReference().equals(sourceKeyFieldName)
+            && ((ColumnReferenceExp) expression).getReference().equals(sourceKeyFieldName),
+        sourcePlanNode
     );
 
     return new ProjectNode(
         new PlanNodeId("Project"),
         sourcePlanNode,
         schema,
         keyFieldName.map(ColumnRef::withoutSource),
-        analysis.getSelectExpressions()
+        sourcePlanNode.getSelectExpressions()
```

Comment:
you can remove this field and get the select expressions from `sourcePlanNode` directly.

The same may be true of other node types.

Reply:
done